packages/toggle-switch-react/src/usePillStyles.ts

```diff
@@ -0,0 +1,31 @@
+import { RefObject, useCallback, useEffect, useState } from "react";
+
+export const usePillStyles = (ref: RefObject<HTMLLabelElement>, shouldTransform: boolean) => {
+    const [width, setWidth] = useState(0);
+    const [distance, setDistance] = useState(0);
+
+    const setStyles = useCallback(() => {
+        if (ref.current && ref.current.parentElement) {
+            setWidth(ref.current.clientWidth);
+
+            const translateDistance = shouldTransform
+                ? ref.current.parentElement.clientWidth - ref.current.clientWidth
+                : 0;
+
+            setDistance(translateDistance);
+        }
+    }, [ref, shouldTransform]);
+
+    useEffect(() => {
+        setStyles();
+        window.addEventListener("resize", setStyles);
```

Comment:
Jeg er litt skeptisk til sånne lyttere på `resize` som også måler dimensjonene til et element, uten noen form for throttle eller debounce. Det ser ut til å funke greit i portalen på min maskin, men det er en klassisk kilde til frame drops på mer komplekse sider.

Hvis du tar en titt i hooken `useElementDimension` så ligger det en throttle-util. Tenker en copy-paste inn her er innafor. Hvis den blir nyttig et tredje sted får vi tenke på noe lurt 😉